docs/README-submodule-api.md

```diff
@@ -0,0 +1,324 @@
+can look like the following:
+
+```json
+[
+  {
+    "type": "image",
+    "url": "docker.io/library/postgres:15.0"
+  },
+  {
+    "type": "image",
+    "url": "docker.io/library/debian:10"
+  }
+]
+```
+
+Parameters:
+* `application_name` -- application name
+
+#### DELETE image
+Allows to remove a given image
+
+Parameters:
+* `image_url` -- a url for image to export. May contain the sha256 (or other sum)
+after `@` sign, in which case we check if the sums match
+* `application_name` -- application name
+
+#### ROLLOUT directory
+Deploys the given composition
+
+Parameters: 
+* `application_name` -- application name
+* `source_directory` -- directory containing the manifests. In case of k8s
+it is a directory where we run apply, or docker-compose up in case of docker-compose.
+
+#### ROLLBACK
+Rolls back the composition to a previous working state
+
+Parameters:
+* `application_name` -- application name
+
+#### PURGE 
+Removes every component of the application
+
+Parameters:
+* `application_name` -- application name
+
+#### STOP
+Stops the application
+
+Parameters:
+* `application_name` -- application name
+
+#### START
+Starts the application
+
+Parameters:
+* `application_name` -- application name
+
+#### ALIVE
+Returns true if application is live
+
+Parameters:
+* `application_name` -- application name
+
+#### HEALTHY
+Returns true if application is healthy and ready to use
+
+Parameters:
+* `application_name` -- application name
+
+### Additional settings
+
+If we think about {{k8s}} we can imagine elements of an application running in different
+namespaces, or requiring some mangling of contexts, or authorization. Instead of creating
+a set of non-portable arguments to, e.g., [ROLLBACK](#rollback) call, we can pass
+the required data in variables, providing them in environment when we call sub-modules.
+When and if needed, we can provide it in the artifact metadata (for non-confidential),
+or inside the artifact in an encrypted manner, using device private key, for instance.
+All this would require some modifications, but is doable, and maybe considered
+for next iterations.
+
+### Persistent storage and configuration
+
+App Update Module needs to store the manifests, both for future reference
+([START](#start)/[STOP](#stop) operations) and [ROLLBACK](#rollback).
+To this end we introduce a `mender-app.conf` in the default location
+`/etc/mender/mender-app.conf`, which will hold all the configuration
+for the update module, and as of the moment of writing this holds
+one setting:
+```
+# cat /etc/mender/mender-app.conf
+persistent_store=/data/mender-app
+```
+
+### App Artifact
+
+#### App Artifact metadata
+
+The Artifact as created by the generator, carries a certain amount of metadata, used to properly
+handle all the above calls.
+
+```json
+{
+  "version": "1.0",
+  "platform": "linux/arm/v7",
+  "application_name": "myapp0",
+  "flavour": "installable-application",
```

Comment:
I think the version is enough; if we'll decide to change/extend implementation we can simply bump the version

Reply:
sure.